qir/qat/TestTools/TestVm.hpp

```diff
@@ -0,0 +1,175 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "qir/qat/Llvm/Llvm.hpp"
+#include "qir/qat/QatTypes/QatTypes.hpp"
+
+#include <stdexcept>
+#include <unordered_map>
+namespace microsoft::quantum
+{
+namespace details
+{
+    inline void initFramework()
+    {
+        static bool was_initialized{false};
+        if (was_initialized)
+        {
+            return;
+        }
+        was_initialized = true;
+
+        llvm::InitializeNativeTarget();
+        llvm::InitializeNativeTargetAsmPrinter();
+    }
+} // namespace details
+class TestProgram
+{
+  public:
+    TestProgram() = default;
+    TestProgram(std::unique_ptr<llvm::Module>&& module, std::unique_ptr<llvm::LLVMContext>&& context)
+      : module_{std::move(module)}
+      , context_{std::move(context)}
+    {
+    }
+    TestProgram(TestProgram&&)      = default;
+    TestProgram(TestProgram const&) = delete;
+
+    ~TestProgram() = default;
+    void parseFromScript(String const& name, String const& script)
+    {
+        context_ = std::make_unique<llvm::LLVMContext>();
+        llvm::SMDiagnostic error;
+        module_ = llvm::parseIR(llvm::MemoryBufferRef(script, name), error, *context_);
+
+        if (!module_)
+        {
+            throw std::runtime_error("Failed to parse script");
+        }
+    }
+
+    llvm::Module& module()
+    {
+        return *module_;
+    }
+
+  private:
+    friend class TestVM;
+    std::unique_ptr<llvm::LLVMContext> context_;
+    std::unique_ptr<llvm::Module>      module_;
+};
+
+class TestVM
+{
+  public:
+    explicit TestVM(TestProgram& program)
+      : program_{program}
+    {
+        details::initFramework();
+    }
+
+    template <typename T> void                   attachGlobalExternalVariable(String const& name, T* pointer);
+    template <typename R, typename X> void       attachRuntimeFunction(String const& name, R (*pointer)(X));
+    template <typename R, typename... Args> void attachRuntimeFunction(String const& name, R (*pointer)(Args...));
+    template <typename R> void                   attachRuntimeFunction(String const& name, R (*pointer)());
+
+    template <typename R, typename... Args> R run(String const& name, Args&&... args);
+    template <typename R> R                   run(String const& name);
+
+  private:
+    void buildVm()
+    {
+        llvm::ExitOnError exit_on_error;
+
+        // TODO(unknown): remove exit_on_error
```

Comment:
What does the `unknown` mean? 
It is unknown who has written this TODO? Or it is unknown which bug/task this TODO relates to?
(more than one occurrence)

Reply:
`unknown` is added by the linter when a username or issue is not present. These TODOs are deprecated anyway, so they should just go.